source/docs/upstream_filters.md

```diff
@@ -0,0 +1,32 @@
+# Converting a downstream filter to be a dual filter
+
+Before coverting a filter to be an upstream filter you should do some basic
+functionality analysis. Make sure
+
+  * The filter does not use any downstream-only functionality, such as clearing
+    cached routes, or performing internal redirects.
```

Comment:
Or anything related to downstream connection? Or things like ext_authz, which I'm not sure how to succinctly explain why it doesn't make sense on upstream. I guess the difference between incoming vs outgoing.

Reply:
you can do things related to the downstream connection.  I could imagine having different auth filters for differed clusters depending on which clusters had access to PII or some such.  One initial use case has to do with doing different header sanitization for "internal" vs" cloud" clusters, where mirroring means you can't configure the correct sanitization per-route, but you might do different sanitization based on "is this a trusted IP" on either path.